arduino/KnexElevator/DataAcquisitionTask.cpp

```diff
@@ -0,0 +1,117 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+// /////////////////////////////////////////////////////////////////////////////
+
+
+// /////////////////////////////////////////////////////////////////////////////
+// Include(s)
+// /////////////////////////////////////////////////////////////////////////////
+#include <Arduino_FreeRTOS.h>
+#include <Arduino.h>
+
+#include "DataAcquisitionTask.h"
+
+
+// /////////////////////////////////////////////////////////////////////////////
+// Local Types
+// /////////////////////////////////////////////////////////////////////////////
+
+
+////////////////////////////////////////////////////////////////////////////////
+DataAcquisitionTask::DataAcquisitionTask(void)
+{
+    // Intentionally left empty
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+DataAcquisitionTask::~DataAcquisitionTask(void)
+{
+    // Intentionally left empty
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+DataAcquisitionTask::Initialize(void)
+{
+    Serial.print("ENTER: ");
+    Serial.println(__PRETTY_FUNCTION__);
+
+    // Fire up our task
+    xTaskCreate(DataAcquisitionTask::RunWrapper,  // Entry Point
+                "DataAcquisitionTask",            // Task Name
+                128,                              // Stack Depth
+                this,                             // Task Parameters
+                2,                                // Priority
+                NULL);                            // Storage for task handle
+
+    Serial.print("LEAVE: ");
+    Serial.println(__PRETTY_FUNCTION__);
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+DataAcquisitionTask::RunWrapper(void * pvParameters)
+{
+    // Entry point to the new task
+    DataAcquisitionTask* thisTask =
+        reinterpret_cast<DataAcquisitionTask*>(pvParameters);
+    thisTask->Run();
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+DataAcquisitionTask::Run(void)
+{
+    Serial.print("ENTER: ");
+    Serial.println(__PRETTY_FUNCTION__);
+
+    while (true)
+    {
+        char timeStampStringBuf[15+1];
+        snprintf(timeStampStringBuf,
+                 sizeof(timeStampStringBuf),
+                 "[%09li]",
+                 micros());
+        Serial.print(timeStampStringBuf);
+        Serial.print(" Hello from ");
+        Serial.println(__PRETTY_FUNCTION__);
+
+        // And print again in a second
+        vTaskDelay(pdMS_TO_TICKS(1000));
```

Comment:
As it turns out, this is a bad way to do time-sensitive delays... this is about 1100 milliseconds.